tests/test_postgresql.py

```diff
@@ -523,8 +523,9 @@ def test_sysid(self):
     def test_save_configuration_files(self):
         self.p.config.save_configuration_files()
 
-    @patch('os.path.isfile', Mock(side_effect=[False, True]))
-    @patch('shutil.copy', Mock(side_effect=IOError))
+    @patch('os.path.isfile', Mock(side_effect=[False, True, False, True]))
+    @patch('shutil.copy', Mock(side_effect=[None, IOError]))
+    @patch('os.chmod', Mock())
```

Comment:
I guess, perfectly, the tests should also check that `chmod`  was actually called with the proper permissions everywhere it is required (where it was added in this PR) 😇